Upstatement/epub.js
# cache locations

## What's in this PR?

Loading locations takes a while, as it loads each section and counts the characters within it. This can take forever on especially long books. However, the array of CFI's that is generated doesn't change, so we should have a way of caching the result and manually setting it. This allows us to get percentages much faster.

## How to test

Load `/highlights` and look in your console. It will take quite some time to load the percentage of the provided CFI on the first load.
Refresh. It should be almost instant.
